core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/utils/EnvelopeTrainPathUtils.kt

```diff
@@ -0,0 +1,55 @@
+package fr.sncf.osrd.sim_infra.utils
+
+import fr.sncf.osrd.envelope_sim.electrification.Electrification
+import fr.sncf.osrd.envelope_sim.electrification.Electrified
+import fr.sncf.osrd.envelope_sim.electrification.Neutral
+import fr.sncf.osrd.envelope_sim.electrification.NonElectrified
+import fr.sncf.osrd.sim_infra.api.Path
+import fr.sncf.osrd.sim_infra.impl.DeadSection
+import fr.sncf.osrd.utils.DistanceRangeMap
+import fr.sncf.osrd.utils.DistanceRangeMapImpl
+import fr.sncf.osrd.utils.units.Distance
+import java.util.function.BiFunction
+
+
+/**
+ * Builds the Electrification Map as a DistanceRangeMap
+ */
+fun buildDistanceElectrificationMap(path: Path): DistanceRangeMap<Electrification?> {
+    var res: DistanceRangeMap<Electrification?> = DistanceRangeMapImpl()
+    res.put(Distance.ZERO, path.getLength(), NonElectrified())
+    res = updateDistanceRangeMaps(
+        res, path.getCatenary()
+    ) { _: Electrification?, catenaryMode: String ->
+        if (catenaryMode == "") NonElectrified() else Electrified(catenaryMode)
+    }
+    res = updateDistanceRangeMaps(
+        res, path.getDeadSections()
+    ) { electrification: Electrification?, deadSection: DeadSection ->
+        Neutral(deadSection.isDropPantograph, electrification)
+    }
+    return res
+}
+
+/**
+ * Return the first map updated with another, using a merge function to fuse the values of intersecting ranges
+ */
+fun <T, U> updateDistanceRangeMaps(
+    map: DistanceRangeMap<T>,
+    update: DistanceRangeMap<U>,
+    updateFunction: BiFunction<T, U, T>
+): DistanceRangeMap<T> {
+    var res = DistanceRangeMapImpl<T>()
+    for ((lower, upper, entry) in map.asList()) {
```

Comment:
```suggestion
    for ((lower, upper, entry) in map) {
```
You can iterate directly